core/src/avm2/activation.rs

```diff
@@ -1232,9 +1241,26 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         method: Gc<'gc, BytecodeMethod<'gc>>,
         index: Index<AbcMultiname>,
     ) -> Result<FrameControl<'gc>, Error> {
-        let multiname = self.pool_multiname(method, index)?;
+        let lookupname = self.pool_lookupname(method, index)?;
```

Comment:
If I'm understanding it right, a Dictionary can only be object-accessed using `AbcMultiname::MultinameL(A)`, right? So maybe instead of adding `LookupName` layer, we do an early check for pooled AbcMultiname's variant? This way the Dictionary (and uint) paths don't need to think about names at all (they can just `pop()` themselves), and heavy-ish Multiname resolving work is delayed till the "normal" path?

Reply:
I should probably alter the namespaces test further to see if an `RTQNameL` also respects object-space lookups...

Reply:
Either way, we don't really care about namespaces or params, so a check on enum variant sounds like a good simple check before branching paths.

Reply:
Test results: `RTQNameL` appears to `VerifyError` if given any non-string name. (This is actually *stricter* than current `master` branch behavior, which will coerce non-strings.)

Special casing this *even further* appears to be viable for `Dictionary`, though I'm still concerned about how viable this is for array-structured objects. I'm pretty sure the following should at least theoretically work, even if it's syntactically forbidden:

```
namespace test_ns = "https://ruffle.rs/test";

class MyArray extends Array {
    //Syntax error, but can be generated via bytecode manipulation
    test_ns var 1 = "test_ns::1";
}

var ma = new MyArray();

ma[1] = "public::1";
trace(ma[1]); //should be public::1
trace(ma.test_ns::[1]); //should be test_ns::1
```

I'll cross that bridge when we get to it, however, and hold off on the `LookupName` split until and unless we need to both care about the namespace set *and* the type of the value on the stack.

Reply:
`LookupName` is now dead